include/xsimd/config/xsimd_arch.hpp

```diff
@@ -61,6 +61,18 @@ namespace xsimd
         {
         };
 
+        template <typename T>
+        constexpr T max_of(T value)
+        {
+            return value;
+        }
+
+        template <typename T, typename... Ts>
+        constexpr T max_of(T head, Ts... tail)
+        {
+            return head > max_of(tail...) ? head : max_of(tail...);
```

Comment:
What about writing this as

```
template <typename T0, typename T1, typename... Ts>
constexpr T max_of(T0 arg0, T1 arg1, Ts... tail) {
  return max_of((arg0 > arg1 ? arg0 : arg1), tail...);
}
```
that would avoid the double call of max_of. Looks good otherwise

Reply:
Thanks, I made that change.